lib/core/src/Cardano/Wallet/Primitive/AddressDiscovery/Sequential.hs

```diff
@@ -244,37 +246,36 @@ defaultAddressPoolGap =
 -------------------------------------------------------------------------------}
 
 data ParentContext (chain :: Role) (key :: Depth -> Type -> Type) where
-    ParentContextUtxoExternal
-        :: key 'AccountK XPub
-        -> ParentContext 'UtxoExternal key
-
-    ParentContextUtxoInternal
-        :: key 'AccountK XPub
-        -> ParentContext 'UtxoInternal key
-
-    ParentContextMultisigScript
-        :: key 'AccountK XPub
+    ParentContextUtxo
+        :: ((key == SharedKey) ~ 'False)
+        => key 'AccountK XPub
+        -> ParentContext chain key
+
+    ParentContextShared
+        :: (key ~ SharedKey)
+        => key 'AccountK XPub
         -> ScriptTemplate
         -> Maybe ScriptTemplate
-        -> ParentContext 'MultisigScript key
+        -> ParentContext chain key
```

Comment:
Suggestion: no need for type parameter here, can be written as:

```hs
    ParentContextShared
        :: SharedKey 'AccountK XPub
        -> ScriptTemplate
        -> Maybe ScriptTemplate
        -> ParentContext chain SharedKey
```

Also, the `chain` parameter seems redundant now?